File: crossecore-java/src/java_/ClassDeclarationBase.java
package java_;
import org.eclipse.emf.common.notify.Notification;
import org.eclipse.emf.common.notify.NotificationChain;
import org.eclipse.emf.common.notify.impl.NotificationImpl;
import org.eclipse.emf.ecore.InternalEObject;
import org.eclipse.emf.ecore.*;
import org.eclipse.emf.ecore.impl.ENotificationImpl;
public class ClassDeclarationBase 
extends TypeDeclarationImpl implements ClassDeclaration
{
	private TypeAccess _superClass;
	public TypeAccess getSuperClass()
	{
		
			return _superClass;
		
		
	}
	public void setSuperClass(TypeAccess value){
		if (value != _superClass) {
			NotificationChain msgs = null;
			if (_superClass != null){
				msgs = ((InternalEObject)_superClass).eInverseRemove(this, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.CLASSDECLARATION_SUPERCLASS, null, msgs);
			}
			if (value != null){
				msgs = ((InternalEObject)value).eInverseAdd(this, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.CLASSDECLARATION_SUPERCLASS, null, msgs);
			}
			msgs = basicSetSuperClass(value, msgs);
			if (msgs != null) {
				msgs.dispatch();
			}
		}
		else if (eNotificationRequired()){
			eNotify(new ENotificationImpl(this, NotificationImpl.SET,Java_PackageImpl.CLASSDECLARATION_SUPERCLASS , value, value));
		}
	}

	
	@Override
	protected EClass eStaticClass() {
		return Java_PackageImpl.Literals.CLASSDECLARATION;
	}
	
	
	public NotificationChain basicSetSuperClass(TypeAccess newobj, NotificationChain msgs) {
		TypeAccess oldobj = _superClass;
		_superClass = newobj;
		if (eNotificationRequired()) {
			ENotificationImpl notification = new ENotificationImpl(this, NotificationImpl.SET, Java_PackageImpl.CLASSDECLARATION_SUPERCLASS, oldobj, newobj);
			if (msgs == null){
				msgs = notification;
			}
			else{
				msgs.add(notification);
			}
		}
		return msgs;
	}
	
	@Override
	public Object eGet(int featureID, boolean resolve, boolean coreType) {
		switch (featureID) {
			case Java_PackageImpl.CLASSDECLARATION_SUPERCLASS:
				return getSuperClass();
		}
		return super.eGet(featureID, resolve, coreType);
	}
	
	@Override
	public void eSet(int featureID, Object newValue) {
		switch (featureID) {
			case Java_PackageImpl.CLASSDECLARATION_SUPERCLASS:
				setSuperClass((TypeAccess) newValue);
				return;
		}
		super.eSet(featureID, newValue);
	}
	
	@Override
	public boolean eIsSet(int featureID) {
		switch (featureID) {
			case Java_PackageImpl.CLASSDECLARATION_SUPERCLASS:
				return _superClass != null; //single != null;
		}
		return super.eIsSet(featureID);
	}
	
	
	
}

File: crossecore-typescript/java_/ClassDeclarationBase.ts
import {EClass} from "ecore/EClass";
import {Set} from "ecore/Set";
import {NotificationChain} from "ecore/NotificationChain";
import {ENotificationImpl} from "ecore/ENotificationImpl";
import {NotificationImpl} from "ecore/NotificationImpl";
import {InternalEObject} from "ecore/InternalEObject";
import {BasicEObjectImpl} from "ecore/BasicEObjectImpl";
import {Java_PackageLiterals} from "./Java_PackageLiterals";
import {ClassDeclaration} from "./ClassDeclaration";
import {TypeAccess} from "./TypeAccess";
import {TypeDeclarationImpl} from "./TypeDeclarationImpl";
		
			export class ClassDeclarationBase
			extends TypeDeclarationImpl
			implements ClassDeclaration
			{
				private _superClass:TypeAccess = null;
				get superClass():TypeAccess{
				
					return this._superClass;
				}
				set superClass(value:TypeAccess) {
					if (value != this._superClass) {
						let msgs:NotificationChain = null;
						if (this._superClass != null){
							msgs = (this._superClass).eInverseRemove(this, BasicEObjectImpl.EOPPOSITE_FEATURE_BASE - Java_PackageLiterals.CLASSDECLARATION_SUPERCLASS, /*null*/ null , msgs);
						}
						if (value != null){
							msgs = value.eInverseAdd(this, BasicEObjectImpl.EOPPOSITE_FEATURE_BASE - Java_PackageLiterals.CLASSDECLARATION_SUPERCLASS, /*null*/ null, msgs);
						}
						msgs = this.basicSetSuperClass(value, msgs);
						if (msgs != null) {
							msgs.dispatch();
						}
					}
					else if (this.eNotificationRequired()){
						this.eNotify(new ENotificationImpl(this, NotificationImpl.SET,Java_PackageLiterals.CLASSDECLARATION_SUPERCLASS , value, value));
					}
				}

			
				public static eStaticClass:EClass;
				
				protected eStaticClass():EClass{
					
					return ClassDeclarationBase.eStaticClass;
				}
			
			
				public basicSetSuperClass(newobj:TypeAccess, msgs:NotificationChain):NotificationChain {
					let oldobj = this._superClass;
					this._superClass = newobj;
					if (this.eNotificationRequired()) {
						let notification = new ENotificationImpl(this, NotificationImpl.SET, Java_PackageLiterals.CLASSDECLARATION_SUPERCLASS, oldobj, newobj);
						if (msgs == null){
							msgs = notification;
						}
						else{
							msgs.add(notification);
						}
					}
					return msgs;
				}
				
			
				public eGet_number_boolean_boolean(featureID:number, resolve:boolean, coreType:boolean):any{
					switch (featureID) {
						case Java_PackageLiterals.CLASSDECLARATION_COMMENTS:
							return this.comments;
						case Java_PackageLiterals.CLASSDECLARATION_ORIGINALCOMPILATIONUNIT:
							return this.originalCompilationUnit;
						case Java_PackageLiterals.CLASSDECLARATION_ORIGINALCLASSFILE:
							return this.originalClassFile;
						case Java_PackageLiterals.CLASSDECLARATION_NAME:
							return this.name;
						case Java_PackageLiterals.CLASSDECLARATION_PROXY:
							return this.proxy;
						case Java_PackageLiterals.CLASSDECLARATION_USAGESINIMPORTS:
							return this.usagesInImports;
						case Java_PackageLiterals.CLASSDECLARATION_ABSTRACTTYPEDECLARATION:
							return this.abstractTypeDeclaration;
						case Java_PackageLiterals.CLASSDECLARATION_ANNOTATIONS:
							return this.annotations;
						case Java_PackageLiterals.CLASSDECLARATION_ANONYMOUSCLASSDECLARATIONOWNER:
							return this.anonymousClassDeclarationOwner;
						case Java_PackageLiterals.CLASSDECLARATION_MODIFIER:
							return this.modifier;
						case Java_PackageLiterals.CLASSDECLARATION_USAGESINTYPEACCESS:
							return this.usagesInTypeAccess;
						case Java_PackageLiterals.CLASSDECLARATION_BODYDECLARATIONS:
							return this.bodyDeclarations;
						case Java_PackageLiterals.CLASSDECLARATION_COMMENTSBEFOREBODY:
							return this.commentsBeforeBody;
						case Java_PackageLiterals.CLASSDECLARATION_COMMENTSAFTERBODY:
							return this.commentsAfterBody;
						case Java_PackageLiterals.CLASSDECLARATION_PACKAGE:
							return this.package;
						case Java_PackageLiterals.CLASSDECLARATION_SUPERINTERFACES:
							return this.superInterfaces;
						case Java_PackageLiterals.CLASSDECLARATION_TYPEPARAMETERS:
							return this.typeParameters;
						case Java_PackageLiterals.CLASSDECLARATION_SUPERCLASS:
							return this.superClass;
					}
					//return this.eGetFromTypeDeclaration(featureID, resolve, coreType);
					return super.eGet(featureID, resolve, coreType);
				}
				
				public eSet_number_any(featureID:number, newValue:any):void {
					switch (featureID) {
						case Java_PackageLiterals.CLASSDECLARATION_SUPERCLASS:
							this.superClass = <TypeAccess> newValue;
							return;
					}
					super.eSet_number_any(featureID, newValue);
				}

				
			}
			
